### ISSUE TITLE
Fix "failed to lazily initialize" warnings in server log

### ISSUE THREAD

**Issue by jasonaowen** [SolutionGuidance/psm#91]
The changes in PR #86 cause warnings to show up in the server log, like:

> 15:45:03,145 WARNING [gov.medicaid.services.impl.ProviderTypeServiceBean] (default task-29) Method output could not be serialized: failed to lazily initialize a collection of role: gov.medicaid.entities.ProviderType.agreementDocuments, could not initialize proxy - no Session (through reference chain: gov.medicaid.entities.SearchResult["items"]->java.util.ArrayList[0]->gov.medicaid.entities.ProviderType["agreementDocuments"])

This is caused by debug-level performance logging - that is, log statements that time how long various lookups take. Lazily-loaded relations, like `ProviderType#agreementDocuments`, are not initialized by default, since the query to look them up can be very expensive. The partially-initialized object throws an exception when you try to access any of its uninitialized fields. The code that wants the `ProviderType` instance doesn't need the `agreementDocuments`, but the logging framework is trying to serialize the instance to JSON, and warning when it fails to do so.

Make these warnings go away. Ideally we could figure out how to serialize only as much of the object as is initialized; less ideal would be to only serialize the non-optional parts, regardless of whether they're loaded; alternatively, we could decide that this kind of performance logging is better done by another method (a library? something built into the application server? a service like new relic?), and replace the logging framework entirely.

**Comment by jasonaowen** [comment]
Thinking about this since opening this issue, and after discussing this with @slifty and @cecilia-donnelly, we decided to simply remove this logging.

We don't have performance requirements at this time, and if we did, I would first try to use some kind of profiling tool before turning up the log levels and reconstructing method execution times from the logs.

Additionally, the logging is done through TopCoder helpers, which we want to remove as part of #214 .

If and when we are working on the performance of HTTP calls, and profiling for some reason does not work, we can reintroduce equivalent logging.
